python/lsst/galaxy_shear/analyzeShearTest.py

```diff
@@ -0,0 +1,277 @@
+                 #   And append to the output fits file
+                 outrec = outSourceCat.addNew()
+                 outrec.set(g1Key, float(g1))
+                 outrec.set(g2Key, float(g2))
+                 outrec.set(filterKey, config.filter)
+                 outrec.set(seeingKey, config.seeing)
+                 outrec.set(shearKey, config.shear_value)
+                 outrec.set(countKey, catCount)
+                 outrec.set(gKey, math.sqrt(g1*g1 + g2*g2))
+                 outrec.set(eAvgKey, eAvg)
+                 outrec.set(e1DevKey, e1Avg-g1)
+                 outrec.set(e2DevKey, e2Avg-g2)
+                 outrec.set(eSumKey, catESum)
+                 outrec.set(e1SumKey, catE1Sum)
+                 outrec.set(e2SumKey, catE2Sum)
+                 outrec.set(weightSumKey, catWeightSum)
+                 nAvgs = nAvgs + 1
+    outSourceCat.writeFits(os.path.join(baseDir, outFile))
+    #   Now summarize the averages and stddevs for all the subfield/epochs
+    if nAvgs > 2:
+        eStddev = math.sqrt( (nAvgs * eSumSq - eSum * eSum)/(nAvgs * (nAvgs - 1)))
+        e1Stddev = math.sqrt((nAvgs * e1DevSumSq - e1DevSum * e1DevSum)/(nAvgs * (nAvgs - 1)))
+        e2Stddev = math.sqrt((nAvgs * e2DevSumSq - e2DevSum * e2DevSum)/(nAvgs * (nAvgs - 1)))
+        print "%d subfields: e=%.4f +-%.4f, e1=%.4f +-%.4f, e2=%.4f +-%.4f, g=%.3f, (%.3f,%.3f)"%(nAvgs,
+               eSum/nAvgs, eStddev, e1DevSum/nAvgs, e1Stddev, e2DevSum/nAvgs, e2Stddev,
+               math.sqrt(g1*g1 + g2*g2), g1, g2)
+    else:
+        print "Not enough subfields to calculate error.  Subfield not added to catalog"
+        print "%d subfields: e = %.4f, e1 = %.4f, e2 = %.4f, g = %.3f"%(nAvgs,
+               eSum/nAvgs, e1DevSum/nAvgs, e2DevSum/nAvgs,
+               math.sqrt(g1*g1 + g2*g2))
+
+    sumSourceCat = afwTable.BaseCatalog(outSourceCat.getSchema())
+    sumrec = sumSourceCat.addNew()
+    sumrec.set(g1Key, float(g1))
+    sumrec.set(g2Key, float(g2))
+    sumrec.set(filterKey, config.filter)
+    sumrec.set(seeingKey, config.seeing)
+    sumrec.set(shearKey, config.shear_value)
+    sumrec.set(countKey, count)
+    sumrec.set(gKey, math.sqrt(g1*g1 + g2*g2))
+    sumrec.set(eAvgKey, eSum/nAvgs)
+    sumrec.set(eStdKey, eStddev)
+    sumrec.set(e1DevKey, e1DevSum/nAvgs)
+    sumrec.set(e1StdKey, e1Stddev)
+    sumrec.set(e2DevKey, e2DevSum/nAvgs)
+    sumrec.set(e2StdKey, e2Stddev)
+    sumrec.set(eSumKey, eSum)
+    sumrec.set(e1SumKey, e1DevSum)
+    sumrec.set(e2SumKey, e2DevSum)
+    sumrec.set(e2SumKey, catE2Sum)
+    sumrec.set(weightSumKey, weightSum)
+    sumSourceCat.writeFits(os.path.join(baseDir, "sum_" + outFile))
+
+if __name__ == "__main__":
+    """
+
+    base_dir         Name of the subdirectory containing run_params file
+    out_file         Name of the catalog for output. .fits is appended
+    test             Name of the test directory, if this is a named test
+    """
```

Comment:
This docstring isn't actually used by Python.  It'd be more normal to just use a code comment here.